src/complex.jl

```diff
@@ -0,0 +1,172 @@
+Base.prevfloat(x::Dual{T,V}) where {T,V<:AbstractFloat} = prevfloat(x.value)
+Base.nextfloat(x::Dual{T,V}) where {T,V<:AbstractFloat} = nextfloat(x.value)
+
+function Base.log(z::Complex{T}) where {A, FT<:AbstractFloat, T<:Dual{A,FT}}
+    T1::T  = 1.25
```

Comment:
```suggestion
    T1  = convert(T, 5//4)
```

Reply:
I am not sure about this, because the compiler cannot optimize the rational number away.

Reply:
This function is copied from Base. One of the goal of this PR is to stay consistent with the implementation in Base.

https://github.com/JuliaLang/julia/blob/575701248adc951038b9727e646230737e5b3c91/base/complex.jl#L564